services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/attachment/AttachmentValidatorTest.kt

```diff
@@ -116,6 +122,49 @@ class AttachmentValidatorTest {
         }
     }
 
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class ValidateExtensionForType {
+        @ParameterizedTest
+        @EnumSource(ApplicationAttachmentType::class)
+        fun `accepts 'pdf' extension for all types`(attachmentType: ApplicationAttachmentType) {
+            AttachmentValidator.validateExtensionForType("file.pdf", attachmentType)
+        }
+
+        @ParameterizedTest
+        @MethodSource("failingExtensions")
+        fun `throws exception when type is other than MUU and extension is not 'pdf'`(
+            attachmentType: ApplicationAttachmentType,
+            extension: String,
+        ) {
+            assertFailure {
+                    AttachmentValidator.validateExtensionForType("file.$extension", attachmentType)
+                }
+                .all {
+                    hasClass(AttachmentInvalidException::class)
+                    messageContains("filename=file.$extension")
+                    messageContains("attachmentType=$attachmentType")
+                }
+        }
+
+        private fun failingExtensions(): List<Arguments> {
+            val extensions = listOf("jpg", "jpeg", "png", "dgn", "dwg", "docx", "txt", "gt")
+
+            return ApplicationAttachmentType.entries
+                .filterNot { it == ApplicationAttachmentType.MUU }
+                .flatMap { type -> extensions.map { Arguments.of(type, it) } }
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = ["jpg", "jpeg", "png", "dgn", "dwg", "docx", "txt", "gt"])
+        fun `accepts other extensions for MUU type attachments`(extension: String) {
+            AttachmentValidator.validateExtensionForType(
+                "file.$extension",
+                ApplicationAttachmentType.MUU
```

Comment:
Also, here would it be better to test "other than LIIKENNEJARJESTELY or VALTAKIRJA" instead of "MUU"?

Reply:
This test will never fail if new attachment types are added because it only tests the MUU type. New tests will need to be added for new types in case they are similar to MUU and allow different file types. And if they are similar to VALTAKIRJA and LIIKENNEJARJESTELY, then no new tests are needed. So, you're right.